app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragment.kt

```diff
@@ -82,11 +83,15 @@ class ConceptCardFragment : InjectableDialogFragment() {
     fun dismissAll(fragmentManager: FragmentManager) {
       val toDismiss = fragmentManager.fragments.filterIsInstance<ConceptCardFragment>()
```

Comment:
It would be really helpful to better understand _why_ ``ConceptCardFragment`` is missing from the fragments that the fragment manager is tracking.

I'm a bit concerned about tracking the fragment itself in this way because shared static state can be really tricky to test and and verify that it's always the correct value at the correct time, so it would be much preferable to try and get the original solution working if we can.

Reply:
Another option might be to try and fetch all fragments by tag, but I'm not sure if that'll handle the concept card stacking situation correctly.

Reply:
Per discussion in meeting, it seems likely that the fragment manager might not be correct for the hints case since it's a dialog fragment (and the fragment manager is handled differently).